dev-lang/vala/vala-0.36.12.recipe

```diff
@@ -0,0 +1,114 @@
+SUMMARY="An object-oriented programming language that uses the GObject system"
+DESCRIPTION="Vala is a new programming language that aims to bring modern \
+programming language features to GNOME developers without imposing any \
+additional runtime requirements and without using a different ABI \
+compared to applications and libraries written in C."
+HOMEPAGE="https://wiki.gnome.org/Projects/Vala"
+COPYRIGHT="2006-2018 Jürg Billeter
+	2006-2018 Raffaele Sandrini"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://download.gnome.org/sources/vala/${portVersion%.*}/vala-$portVersion.tar.xz"
+CHECKSUM_SHA256="b4081d3d313cf9c3e8d7d795983cf8f7ed99286d1abc1b5709b0f17883387cdb"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+portVers="${portVersion%.*}"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersionCompat
+	cmd:vala$secondaryArchSuffix
+	cmd:vala_$portVers$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_gen_introspect_$portVers$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valac_$portVers$secondaryArchSuffix
+	cmd:vapicheck$secondaryArchSuffix
+	cmd:vapicheck_$portVers$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vapigen_$portVers$secondaryArchSuffix
+	lib:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgobject_2.0$secondaryArchSuffix
+	lib:libdbus_1$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+REPLACES="
+	vala${secondaryArchSuffix}_common
+	"
+
+PROVIDES_devel="
+	vala${secondaryArchSuffix}_devel = $portVersionCompat
+	"
+REQUIRES_devel="
+	vala$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgobject_2.0$secondaryArchSuffix
+	devel:libdbus_1$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:awk
+	cmd:bison
+	cmd:cmp
+	cmd:diff
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage vala$secondaryArchSuffix \
+	$binDir/vala \
+	$binDir/valac \
+	$binDir/vapicheck \
+	$binDir/vapigen \
+	$binDir/vala-$portVers \
+	$binDir/valac-$portVers \
+	$binDir/vapicheck-$portVers \
+	$binDir/vapigen-$portVers
+
+BUILD()
+{
+	autoreconf -fi
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm -f $libDir/*.la
+
+	fixPkgconfig
```

Comment:
If you insert this:
```diff
+	prepareInstalledDevelLib libvala-$portVers
 	fixPkgconfig
```
then you can add a `devel:libvala_$portVers` in `PROVIDES_devel`:
```diff
 PROVIDES_devel="
 	vala${secondaryArchSuffix}_devel = $portVersionCompat
+	devel:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
 	"
```

I think this is what we should do, but I might be wrong.

Reply:
Checking ...

Reply:
ok, this seems to work +1 but I also have vapigen.pc and vapigen-0.36.pc for the pkg-config files in the package, if I look at the content it looks ok, but "pkg-config-x86 vapigen --cflags" (--libs) doesn't produce any output?

Reply:
Did you install both the base package and the devel subpackage?
If you did not install the devel subpackage then pkg-config will not find some files it needs.
BTW, Debian uses a very granular packaging: For vala, they have several binary pacakges, and one of them only has the shared lib. So I guess it would be safe to only have the shared lib in the base package.
Well, for now we need to keep the `cmd:` in the base package.

Hmm, it looks like my idea of moving some data files to the devel package is not appripriatte. (Sorry!)
After all, if a recipe has `cmd:vapigen` (and `cmd:pkg_config`) in its `BUILD_PREREQUIRES` then pkg-config and vapigen should be usable with no other dependencies.

It looks like we really need to check that the split is OK before the merge.
I did not do it, (but I did mention that I was not sure about the move :-)
So, sorry again for the bad suggestions, if it turns out that the moves are a bad idea.

Reply:
OK.
So the question here is:
_Is it OK if `pkg-config-x86 vapigen --libs` does not produce any output?_
Good question.
(I don't know the answer.)

Regading the data files, it looks like the vapi files are required by valac.
I wonder if, after all, we should not just drop the devel package and put everything (except the debuginfo) in the base package.

Reply:
Are any vala experts reading this thread?
What would you recommend, in particular, about the split?

Reply:
It turns out that we never had a devel subpackage for vala. The only subpackage we've had is the vala_common.
So, shall we just drop these blocs?
```
PROVIDES_devel="
	vala${secondaryArchSuffix}_devel = $portVersionCompat
	devel:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
	"
REQUIRES_devel="
	vala$secondaryArchSuffix == $portVersion base
	"
```
and
```
	prepareInstalledDevelLib libvala-$portVers
	fixPkgconfig

	# devel package
	packageEntries devel \
		$developDir \
		$dataDir/aclocal \
		$dataDir/vala \
		$dataDir/vala-$portVers
```

Reply:
Even after removing the devel package output is still the same ...

Reply:
This is probably a bit exessive, but at least we know that we won't have any issues caused by an incorrect split.
BTW, we don't even need to have any `REPLACES` here, since any past vala_devel would be automatically removed when the user will upgrade vala.

Reply:
Rebuilding from scratch to check ...

Reply:
Well, in that case I think we can assume that it is normal to have no output for `pkg-config-x86 vapigen --libs`.
What this means, if I'm not mistaken, is that GUI apps built with vala don't need libvala.
I guess the shared lib is only used by the runtimes provided by vala itself.

Reply:
Ok, build is still ok (withouth devel package) but out for "pkg-config-x86 vapigen --cflags" still doesn't produce any output ...

Reply:
Well, I think this is normal.
I'd say that when a GUI app is built with vala, the only thing that is needed at build time are the headers.

BTW, I just realized that we need to keep the call to `fixPkgconfig`.
(I'm checking this right now.)
Without this, the `*.pc` files would be in `/boot/system/lib/pkgconfig/`, but we need them to be in `/boot/system/develop/lib/pkgconfig/` otherwise pkgconfig won't find them.

Reply:
It looks like all we need in `INSTALL()` is:
```
INSTALL()
{
	make install

	rm -f $libDir/*.la

	fixPkgconfig
}
```